phpunit/style-engine/class-wp-style-engine-css-declarations-test.php

```diff
@@ -121,22 +121,24 @@ public function test_should_strip_html_tags_and_remove_unsafe_css_properties() {
 
 
 	/**
-	 * Tests that calc, clamp, min, max, and minmax CSS functions are allowed.
+	 * Tests that calc, clamp, min, max, minmax CSS functions and --wp--* CSS custom properties are allowed.
 	 *
 	 * @covers ::get_declarations_string
 	 * @covers ::filter_declaration
 	 */
 	public function test_should_allow_css_functions_and_strip_unsafe_css_values() {
 		$input_declarations                        = array(
-			'background'       => 'var(--wp--preset--color--primary, 10px)', // Simple var().
-			'font-size'        => 'clamp(36.00rem, calc(32.00rem + 10.00vw), 40.00rem)', // Nested clamp().
-			'width'            => 'min(150vw, 100px)',
-			'min-width'        => 'max(150vw, 100px)',
-			'max-width'        => 'minmax(400px, 50%)',
-			'padding'          => 'calc(80px * -1)',
-			'background-image' => 'url("https://wordpress.org")',
-			'line-height'      => 'url("https://wordpress.org")',
-			'margin'           => 'illegalfunction(30px)',
+			'background'                        => 'var(--wp--preset--color--primary, 10px)', // Simple var().
+			'font-size'                         => 'clamp(36.00rem, calc(32.00rem + 10.00vw), 40.00rem)', // Nested clamp().
+			'width'                             => 'min(150vw, 100px)',
+			'min-width'                         => 'max(150vw, 100px)',
+			'max-width'                         => 'minmax(400px, 50%)',
+			'padding'                           => 'calc(80px * -1)',
+			'background-image'                  => 'url("https://wordpress.org")',
+			'line-height'                       => 'url("https://wordpress.org")',
+			'margin'                            => 'illegalfunction(30px)',
+			'--wp--style--unstable-gallery-gap' => '12em',
```

Comment:
Yeah I assume so, since it should now work in Core any way. 

But now that I look at it, it doesn't really match the subject of the test: `test_should_allow_css_functions_and_strip_unsafe_css_values`

Also, it's very specific to the changes in this PR. Maybe I should add a separate test to make that obvious. Thank you for the suggestion!

Reply:
> we'll still want this line when eventually backporting to core, right? (So we're testing that setting custom properties works) Assuming so, this looks good to include.

Actually, I was just adding the new test and the body of it is the same as `test_should_allow_css_functions_and_strip_unsafe_css_values`, and I expect that in any Core patch review we'd be ask to combine them regardless 😄 

And you're right: we'll probably want a test like this in Core anyway given that it supports CSS properties.

So I might just rename the test to: `test_should_allow_css_functions_and_custom_properties_and_strip_unsafe_css_values`

And see what happens when we next backport 😄 

Does that sound okay?

Reply:
> Does that sound okay?

Sounds good to me!